### ISSUE TITLE
How to predict a test set of images?

### ISSUE THREAD

**Issue by canyilmaz90** [qqwweee/keras-yolo3#380]
Using yolo_video.py we can get results real-time, but is there any function to evaluate our model on thousands of images? I tried to read codes, unfortunately I couldn't find any function including something like model.predict() or model.evaluate(). 

If someone knows something about this, please let me know.

**Comment by bluesy7585** [comment]
I had same question. but couldn't find the answer.
so I made some changes to the code. it process images in folder and output txt files.
python yolo_video.py --imgdir .\sample_folder --txt
https://github.com/bluesy7585/keras-yolo3

**Comment by canyilmaz90** [comment]
> I had same question. but couldn't find the answer.
> so I made some changes to the code. it process images in folder and output txt files.
> python yolo_video.py --imgdir .\sample_folder --txt
> https://github.com/bluesy7585/keras-yolo3

@bluesy7585 thank you for your reply. Have you modified only the "yolo.py" and "yolo_video.py" scripts?

**Comment by bluesy7585** [comment]
@canyilmaz90 
I mode some changes in other files as well, but it only effects training phase.

**Comment by canyilmaz90** [comment]
@bluesy7585 this worked very well for my case, thanks a lot!

**Comment by cuimiao187561** [comment]
How to evaluate the test set, if you have already solved it, can you send the code to me?(cuimiao188@163.com) @canyilmaz90

**Comment by canyilmaz90** [comment]
@cuimiao187561 look at the solution proposed by @bluesy7585 , codes are available in his/her repo and its very easy to implement.

**Comment by gzz1529657064** [comment]
> I had same question. but couldn't find the answer.
> so I made some changes to the code. it process images in folder and output txt files.
> python yolo_video.py --imgdir .\sample_folder --txt
> https://github.com/bluesy7585/keras-yolo3

I read your code.But after using your code, executing the command will still give an error. In my yolo_video.py, I can detect video. 
"python yolo_video.py --imgdir ./detection_results/test_image/ --txt"
./detection_results/test_image/  is my test image folder    
executing command error!     
This is my WeChat: wxid_cbjyy2ydd0j422      May I ask you some questions? Thanks!

**Comment by IrisDinge** [comment]
> > I had same question. but couldn't find the answer.
> > so I made some changes to the code. it process images in folder and output txt files.
> > python yolo_video.py --imgdir .\sample_folder --txt
> > https://github.com/bluesy7585/keras-yolo3
> 
> I read your code.But after using your code, executing the command will still give an error. In my yolo_video.py, I can detect video.
> "python yolo_video.py --imgdir ./detection_results/test_image/ --txt"
> ./detection_results/test_image/ is my test image folder
> executing command error!
> This is my WeChat: wxid_cbjyy2ydd0j422 May I ask you some questions? Thanks!

Hey bro, if u got some reply, would u mind to share with me?

**Comment by IrisDinge** [comment]
> > I had same question. but couldn't find the answer.
> > so I made some changes to the code. it process images in folder and output txt files.
> > python yolo_video.py --imgdir .\sample_folder --txt
> > https://github.com/bluesy7585/keras-yolo3
> 
> I read your code.But after using your code, executing the command will still give an error. In my yolo_video.py, I can detect video.
> "python yolo_video.py --imgdir ./detection_results/test_image/ --txt"
> ./detection_results/test_image/ is my test image folder
> executing command error!
> This is my WeChat: wxid_cbjyy2ydd0j422 May I ask you some questions? Thanks!

oh, i fixed it sorry for disturbing!
